File: Sources/GatewayCore/LocalModelProfile.swift
import Foundation

/// One local language-model identity Gateway Forge relies on.
///
/// The base model is shared, but the identities are not interchangeable:
/// `gateway-composer` works from documented material and drafts or curates a
/// session, while `gateway-cartographer` is allowed to read only the listener's
/// own contemporaneous visits. Keeping the required set here gives Setup, the
/// evaluator and readiness one inventory instead of letting each remember a
/// different subset.
public struct LocalModelProfile: Equatable, Sendable {
    public var model: String
    public var modelfile: String

    public init(model: String, modelfile: String) {
        self.model = model
        self.modelfile = modelfile
    }
}

public enum LocalModelProfiles {
    public static let required: [LocalModelProfile] = [
        LocalModelProfile(model: Compose.model, modelfile: "Modelfile"),
        LocalModelProfile(model: Cartographer.model, modelfile: "Cartographer.modelfile"),
    ]

    public static var models: [String] { required.map(\.model) }
}

File: cross-platform/src/core/localModelProfile.ts
/**
 * The local language-model identities Gateway Forge relies on, ported from
 * `LocalModelProfile.swift`.
 *
 * The base model is shared, but the identities are not interchangeable:
 * the composer works from documented material and drafts or curates a
 * session, while the cartographer is allowed to read only the listener's
 * own contemporaneous visits.
 */
import { composeModel } from "./compose.js";
import { cartographerModel } from "./cartographer.js";

export interface LocalModelProfile { model: string; modelfile: string }

export const requiredLocalModelProfiles: LocalModelProfile[] = [
  { model: composeModel, modelfile: "Modelfile" },
  { model: cartographerModel, modelfile: "Cartographer.modelfile" },
];

export const requiredLocalModels: string[] = requiredLocalModelProfiles.map(p => p.model);
